src/linux/init/config.cpp

```diff
@@ -576,6 +515,11 @@ Return Value:
 
     wsl::linux::WslDistributionConfig Config{CONFIG_FILE};
 
+    if (getenv(LX_WSL2_SYSTEM_DISTRO_SHARE_ENV) != nullptr)
+    {
+        Config.GuiAppsEnabled = true;
+    }
```

Comment:
Why is this needed?

Reply:
That's because we need `GuiAppsEnabled` to be set before we decide whether to create a systemd user generator or not. The Windows side setting is set in mini_init, which will only set `LX_WSL2_SYSTEM_DISTRO_SHARE_ENV` if gui apps are enabled